src/variable.js

```diff
@@ -89,11 +95,16 @@ function variable_define(name, inputs, definition) {
   }
   return variable_defineImpl.call(this,
     name == null ? null : String(name),
-    inputs == null ? [] : map.call(inputs, this._module._resolve, this._module),
+    inputs == null ? [] : map.call(inputs, this._resolve, this),
     typeof definition === "function" ? definition : constant(definition)
   );
 }
 
+function variable_resolve(input) {
+  if (this._shadows.has(input)) return this._shadows.get(input);
+  return this._module._resolve(input);
+}
```

Comment:
Just to make sure that `input` is the _name_ of an input and not a variable, and for consistency with `module_resolve`, I’d prefer to use `name` here. Also, stylistic preference to use nullish coalescing for brevity.

```suggestion
function variable_resolve(name) {
  return this._shadows.get(name) ?? this._module._resolve(name);
}
```